### PR TITLE
Use precompiled state machine binary to run canister tests

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR introduces the precompiled state machine binary to run canister integration tests. This greatly speeds up the CI pipeline.

Note: the crate `state_machine_client` will be moved to a separate repository in the future and published on crates.io.

### CODE REVIEW

File: src/state_machine_client/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
```
Comment (nmattia) [inline src/state_machine_client/Cargo.toml:1]: can we also get a README in this directory?

File: src/canister_tests/src/framework.rs [hunk 2]
```diff
@@ -135,14 +134,39 @@ fn get_wasm_path(env_var: String, default_path: &path::PathBuf) -> Option<Vec<u8
...
 
-/* Here are a few useful helpers for writing tests */
-pub fn install_empty_canister(env: &StateMachine) -> CanisterId {
-    env.install_canister(EMPTY_WASM.clone(), vec![], None)
-        .expect("failed to install empty canister.")
+/** The Wasm module for the current II build, i.e. the one we're testing */
+pub static STATE_MACHINE_BINARY: &str = "../../ic-test-state-machine";
+
+pub fn env() -> StateMachine {
+    let path = match env::var_os("STATE_MACHINE_BINARY") {
```
Comment (nmattia) [inline src/canister_tests/src/framework.rs:141]: shouldn't `state_machine_client` deal with this?

**Review by nmattia — APPROVED** [review]
mostly rubber stamping

File: src/canister_tests/src/framework.rs (hunk 2 above)
Comment (frederikrothenberger) [reply src/canister_tests/src/framework.rs:141]: Hm, I'm not convinced. The library should not keep where to find the binary or which `env` variable can be supplied insted...

Comment (frederikrothenberger) [reply src/canister_tests/src/framework.rs:141]: We can still move this code later, if you feel strongly about it.

**Pull request merged**